src/geometry.rs

```diff
@@ -0,0 +1,137 @@
+//! Structures for creating and storing geometric primitives.
+use mint;
+use std::collections::HashMap;
+use genmesh::{Vertex as GenVertex, EmitTriangles, Triangulate};
+use genmesh::generators::{self, IndexedPolygon, SharedVertex};
+
+/// A shape of geometry that is used for mesh blending.
+#[derive(Clone, Debug)]
+pub struct Shape {
+    /// Vertices.
+    pub vertices: Vec<mint::Point3<f32>>,
+    /// Normals.
+    pub normals: Vec<mint::Vector3<f32>>,
+    /// Tangents.
+    pub tangents: Vec<mint::Vector4<f32>>,
+    /// Texture co-ordinates.
+    pub tex_coords: Vec<mint::Point2<f32>>,
+}
+
+impl Shape {
+    /// Create an empty shape.
+    pub fn empty() -> Self {
+        Shape {
+            vertices: Vec::new(),
```

Comment:
FYI, `vec![]` does the same thing. It's faster to type. No need to change though.

Reply:
I know, but it's not my code actually. And I prefer `Vec::new()` as it more verbose.
There is no macro for `HashMap::new`, for example, so using `Vec::new` makes code a little bit more "solid"

Reply:
:+1: